micrometer-core/src/main/java/io/micrometer/core/instrument/binder/jvm/JvmThreadMetrics.java

```diff
@@ -107,4 +103,41 @@ private static String getStateTagValue(Thread.State state) {
         return state.name().toLowerCase().replace("_", "-");
     }
 
+    private Long getThreadCountByState(Thread.State state) {
+        // If the thread-local is empty, this will also trigger populating it
+        // i.e.: getting a new snapshots for thread counts by thread state.
+        Map<Thread.State, Long> threadCountsByState = threadCountsByStateThreadLocal.get();
+        Long count = threadCountsByState.remove(state);
+
+        // This means that this state was already queried. Assuming that a Gauge will be
+        // queried once per publication, this should mean that we need a new snapshot.
```

Comment:
There's no guarantee about how many times a gauge is queried per step interval. Many registry implementations would only have need to query once per step, but for example, a Prometheus setup with multiple Prometheus servers scraping the same instances for redundancy would read each gauge more than once per step.

Reply:
Multiple reads will not be a problem, will trigger the expiration of the snapshot, as well as re-create the snapshot, performance will return to the same as before, however, this is the extreme case I think, in the vast majority of cases, this commit is to reduce repeated calls and optimize performance.

Reply:
@shakuzen Do you have any other comments on this?